lib/cacheblockstore/longtail_cacheblockstore.c

```diff
@@ -906,6 +961,22 @@ static int CacheBlockStore_GetStats(struct Longtail_BlockStoreAPI* block_store_a
     return 0;
 }
 
+static int CacheBlockStore_Flush(struct Longtail_BlockStoreAPI* block_store_api, struct Longtail_AsyncFlushAPI* async_complete_api)
+{
+    LONGTAIL_LOG(LONGTAIL_LOG_LEVEL_DEBUG, "CacheBlockStore_Flush(%p, %p)", block_store_api, async_complete_api)
+    struct CacheBlockStoreAPI* cacheblockstore_api = (struct CacheBlockStoreAPI*)block_store_api;
+    if (cacheblockstore_api->m_PendingRequestCount > 0)
+    {
+        Longtail_LockSpinLock(cacheblockstore_api->m_Lock);
```

Comment:
We can get into trouble here if m_PendingRequestCount goes to zero from the time we check to when we add our listener to the list of pending notifiers...